source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -164,15 +177,30 @@ void HeaderToMetadataFilter::writeHeaderToMetadata(Http::HeaderMap& headers,
                                                    Http::StreamFilterCallbacks& callbacks) {
   StructMap structs_by_namespace;
 
-  for (const auto& rulePair : rules) {
-    const auto& header = rulePair.first;
-    const auto& rule = rulePair.second;
+  for (const auto& rule : rules) {
+    const auto& header = rule.header();
+    const auto& proto_rule = rule.rule();
     const Http::HeaderEntry* header_entry = headers.get(header);
 
-    if (header_entry != nullptr && rule.has_on_header_present()) {
-      const auto& keyval = rule.on_header_present();
-      absl::string_view value = keyval.value().empty() ? header_entry->value().getStringView()
-                                                       : absl::string_view(keyval.value());
+    if (header_entry != nullptr && proto_rule.has_on_header_present()) {
+      const auto& keyval = proto_rule.on_header_present();
+      std::string rewritten_value;
```

Comment:
maybe note here that we do this so that we can bind `value` to `rewritten_value` without the string going out of scope